src/routes/safe/components/Balances/index.tsx

```diff
@@ -47,12 +47,13 @@ const Balances = (props) => {
 
   const address = useSelector(safeParamAddressFromStateSelector)
   const featuresEnabled = useSelector(safeFeaturesEnabledSelector)
+  const currentPath = history.location.pathname
 
   useFetchTokens()
 
   useEffect(() => {
-    const showCollectibles = COLLECTIBLES_LOCATION_REGEX.test(history.location.pathname)
-    const showCoins = COINS_LOCATION_REGEX.test(history.location.pathname)
+    const showCollectibles = COLLECTIBLES_LOCATION_REGEX.test(currentPath)
```

Comment:
Why are we not using `react-router` for that?

Reply:
Because is not simplest as just display the collectibles/coins tab on URL changes. We also need to verify certain conditions to allow the user to nav there:
```
    if (!showCollectibles && !showCoins) {
      history.replace(`${SAFELIST_ADDRESS}/${address}/balances`)
    }

    const erc721Enabled = featuresEnabled && featuresEnabled.includes('ERC721')

    if (erc721Enabled) {
      subMenuOptions.push({
        enabled: showCollectibles,
        legend: 'Collectibles',
        url: `${SAFELIST_ADDRESS}/${address}/balances/collectibles`,
      })
    } else {
      if (showCollectibles) {
        history.replace(subMenuOptions[0].url)
      }
    }
```

Reply:
Honestly I don't see any difficulties in implementing this logic with `react-router`. You just use `<Switch>` and `<Route>` components and place the logic for rendering inside `Route`'s component `render` prop

Similarly how it's done in `src/routes/safe/components/Layout/index.tsx` and `src/routes/index.tsx`

Reply:
I tried to implement that but it just doesn't make sense, the logic is also used for display or hide the tabs coins and collectibles based on the current contract version. Adding the logic of the route will repeat unnecessary logic

Reply:
```
  useEffect(() => {
    const subMenuOptions = [{ legend: 'Coins', url: `${SAFELIST_ADDRESS}/${address}/balances` }]

    const erc721Enabled = featuresEnabled && featuresEnabled.includes('ERC721')

    if (erc721Enabled) {
      subMenuOptions.push({
        legend: 'Collectibles',
        url: `${SAFELIST_ADDRESS}/${address}/balances/collectibles`,
      })
    }

    setState((prevState) => ({
      ...prevState,
      erc721Enabled,
      subMenuOptions,
    }))
  }, [featuresEnabled, address])

<Route
  render={() => {
    if (!erc721Enabled) {
          return <Redirect />
    }
    
    return <Collectibles />
/>
```

I don't see how it doesn't make sense. This is all placed in one file so the logic is reusable